File: sample-projects/php/src/Domain/Model/NoSuchCreatureException.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Domain\Model;

use RuntimeException;

final class NoSuchCreatureException extends RuntimeException
{
    public function __construct(CreatureId $id)
    {
        parent::__construct('No such creature in the dungeon: ' . $id->idAsString());
    }
}

File: sample-projects/rust/src/domain/model/no_such_creature_exception.rs
use std::error::Error;
use std::fmt;

use crate::domain::model::creature_id::CreatureId;

#[derive(Debug)]
pub struct NoSuchCreatureException {
    id: CreatureId,
}

impl NoSuchCreatureException {
    pub fn new(id: CreatureId) -> Self {
        NoSuchCreatureException { id }
    }
}

impl fmt::Display for NoSuchCreatureException {
    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(formatter, "No such creature in the dungeon: {}", self.id.id)
    }
}

impl Error for NoSuchCreatureException {}
